src/wallet/wallet2.cpp

```diff
@@ -3084,6 +3289,10 @@ crypto::hash8 wallet2::get_short_payment_id(const pending_tx &ptx) const
 // take a pending tx and actually send it to the daemon
 void wallet2::commit_tx(pending_tx& ptx)
 {
+  // allow sending to subaddresses only on testnet
+  if (ptx.construction_data.dest_subaddr && !testnet())
+    throw std::runtime_error("Sending to subaddresses is allowed only on testnet at the moment!");
```

Comment:
I'm not entirely sure. I think it's a good idea to ask the community volunteers to test this scheme before deploying it in the real world. When introducing new features, how should the testing be done in general? I guess the multisig PR is in a pretty similar situation.